Implement Action Link New Variation for Popular Section

MartinJJones opened the pull request:
## What

Update the popular links section for the new homepage design, this work involved:

- Create new view `popular_links` view template
- Updating the popular links to use the [new variation of the action-link component](https://components.publishing.service.gov.uk/component-guide/action_link/with_light_icon/preview)
- Move existing CSS from `homepage.scss` and create 2 new stylesheets, `links_and_search` and `popular_links`, we then include the required stylesheet depending on the template used (existing or new design)
- Added a temporary override to increase the action link icon size on desktop, this will be moved back into the gem in the future.

## Why

This is part of changes that will be gradually rolled out to the homepage. As these changes are not going to be made live immediately, I have attempted to split the code in a way that will allow them to be easily toggled by the new_design parameter.

[Trello card](https://trello.com/c/2Xocwwne/2103-look-and-feel-homepage-implement-action-link-new-variation-for-popular-section-m), [Jira issue NAV-8371](https://gov-uk.atlassian.net/browse/NAV-8371)

## Visual Changes

### Existing design - mobile

<img width="485" alt="popular-links-mobile-before" src="https://github.com/alphagov/frontend/assets/28779939/1dd85662-67c7-46cb-abd9-8f8449bdaf05">

### New design - mobile

<img width="495" alt="popular-links-mobile-after" src="https://github.com/alphagov/frontend/assets/28779939/0d4e3734-49dd-476f-9125-900bcb3d6dee">

### Existing design - desktop

<img width="1196" alt="popular-links-desktop-before" src="https://github.com/alphagov/frontend/assets/28779939/f0d4eae2-8008-45df-aff2-2ebeee8c0578">

### New design - desktop

<img width="1039" alt="popular-links-desktop-after" src="https://github.com/alphagov/frontend/assets/28779939/85b39f91-f46d-4855-be5a-c832d9961ba9">

---
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

MartinJJones committed:
Create new popular_links partial

Create a new popular_links partial ready to be used in the new homepage design.

- Create new view `popular_links` view template
- Move existing CSS from `homepage.scss` and create 2 new stylesheets, `links_and_search` and `popular_links`, we then include the required stylesheet depending on the template used (existing or new design)

MartinJJones committed:
Use new popular_links view if new_design is true

patrickpatrickpatrick commented:
This looks good to me. Looks like the design and splits code in a logical way which will make making the changes permanent more straightforward.

MartinJJones committed:
Update popular links following design feedback

- Ensure font-size for popular links is always 19px, mobile or desktop
- Update padding to match the required design

We may be able to remove some of the padding styles by using `gap`, `row-gap` or `column-gap`.

patrickpatrickpatrick commented:
Just to say; still looks good.
